### ISSUE TITLE
Run eel exposed function from another file

### ISSUE THREAD

**Issue by 04fsnape** [ChrisKnott/Eel#150]
Hello,

I am trying to execute a function that I have written in JavaScript and exposed to eel, but from another python file.

At the moment, I am just doing

```
import gui


gui.run_test_function("x")
```

However, that results in

> Traceback (most recent call last):
>   File "main.py", line 328, in \<module>
>     main()
>   File "main.py", line 314, in main
>     gui.run_test_function("aaaaaa")
>   File "../interface\gui.py", line 102, in run_test_function
>     eel.test_function(test_text)
> AttributeError: module 'eel' has no attribute 'test_function'

However, if I run it from within the `gui.py` folder, it works perfectly.

How can I fix this?

Thanks.

**Comment by samuelhwilliams** [comment]
Please can you share a link to your git repo/code or otherwise include more details about your directory structures, files, and code. This will help with debugging. It looks like you might be importing the wrong `eel` into your `gui.py` folder - but hard to say for sure without more information.

**Comment by 04fsnape** [comment]
@samuelhwilliams 

Here is my file structure

```
interface
    node_modules
    web
        css
            main.css
        html
            index.html
        images
            ...
        js
            ...
        themes
            ...
    gui.py
    main.js
    package.json
    package-lock.json
    renderer.js
src
    main.py
```

Here is the relevant part of my `gui.py` script
```
import eel


options = {
	'mode': 'custom',
	'args': ['node_modules/electron/dist/electron.exe', '.']
}

def run_test_function(test_text):
	eel.test_function(test_text)


if __name__ == "__main__":
	open("console.txt", "w").close()
	eel.init('web')
	eel.start('html/index.html', options=options)
```

And `main.py`:
```
sys.path.insert(0, '../interface')
import gui

gui.run_test_function("test")
```

And here's a script in my `index.html` file
```
<script type="text/javascript">
        eel.expose(test_function);
        function test_function(test_text) {
          alert(test_text);
        }
</script>
```

**Comment by samuelhwilliams** [comment]
Ok, the problem is in how you're executing the files and your understanding of the `__name__` variable.

The file that you execute is the entrypoint to the program, and within that file, `__name__` resolves to `__main__`. In other files/modules/etc that get imported into the program, `__name__` resolves to the name of that file. This allows Python scripts to work out which file was the entrypoint to the program.

When you run `gui.py` directly, within that file `__name__` resolves to `__main__`.  Because of this, the conditional block of code in that file gets run - initialising eel and starting it running.

When you run `main.py` instead, the `__name__` reference in `gui.py` now resolves to just the word `gui.py` (or potentially some path leading to that file). So the conditional block doesn't get run when you import that module, which means eel never gets initialised, and so never communicates with javascript to work out what functions are available.

TL;DR I think you need to initialise eel in `main.py` before using the `gui.run_test_function` function.

Let me know if this helps - I haven't actually tested this.

**Comment by 04fsnape** [comment]
@samuelhwilliams I understand. I fixed it by merging the two files. Thanks for your help.
